lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -445,24 +445,30 @@ namespace Js
             if (recycler->ReportExternalMemoryAllocation(length))
             {
                 buffer = (BYTE*)allocator(length);
-                if (buffer == nullptr)
+            }
+
+            if (buffer == nullptr)
+            {
+                recycler->CollectNow<CollectOnTypedArrayAllocation>();
+                if (recycler->ReportExternalMemoryAllocation(length))
```

Comment:
I also don't understand and still don't understand why you are not reporting twice. Once on line 445, then allocation failed, collect, and report again on line 453.